connectivity/check/context.go

```diff
@@ -492,8 +492,8 @@ func (ct *ConnectivityTest) DetectMinimumCiliumVersion(ctx context.Context) (*se
 		if err != nil {
 			return nil, fmt.Errorf("unable to fetch cilium version on pod %q: %w", name, err)
 		}
-		v, _, _ := strings.Cut(stdout.String(), "-") // strips proprietary -releaseX suffix
-		podVersion, err := semver.Parse(v)
+		v, _, _ := strings.Cut(stdout.String(), "-")                      // strips proprietary -releaseX suffix
+		podVersion, err := semver.Parse(strings.Replace(v, "\n", "", -1)) // strips newline
```

Comment:
I don't know where this is coming from but the remaining part looks great. Thanks!

Reply:
it's coming from my installation with master, the cilium version std output is 1.12.90\n, and the above cut command didn't remove any prefix.

```
$ ./cilium connectivity test --force-deploy
ℹ️  Single-node environment detected, enabling single-node connectivity test
connectivity test failed: unable to parse cilium version on pod "cilium-prz84": Invalid character(s) found in patch number "90\n"
```

Reply:
I've sent #1016 which fixes this in a slightly more robust way.

Reply:
Thanks, I will rebase once #1016 lands in :100: